src/v3.rs

```diff
@@ -113,6 +124,25 @@ impl Sender {
         Sender { api_key }
     }
 
+    #[cfg(feature = "async")]
+    pub fn send(&self, mail: &Message) -> impl Future<Item=Response, Error=SendgridError> {
```

Comment:
I think there's an opportunity to refactor this further. We could consolidate the header construction into a function and then the `send` implementation can just be the configured client. This will cut down on the code duplication.